File: src/types/holiday.ts
export interface Holiday {
  id: number;
  schoolYear: string;
  state: string;
  name: string;
  startDate: string;
  endDate: string;
  createdAt: string;
}

export interface HolidayPeriod {
  name: string;
  startDate: string;
  endDate: string;
}

export interface HolidayImportData {
  schoolYear: string;
  state: string;
  holidays: HolidayPeriod[];
}

File: src-tauri/src/models/holiday.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Holiday {
    pub id: i64,
    pub school_year: String,
    pub state: String,
    pub name: String,
    pub start_date: String,
    pub end_date: String,
    pub created_at: String,
}

/// Struktur für den JSON-Import einer einzelnen Ferienperiode
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HolidayPeriod {
    pub name: String,
    pub start_date: String,
    pub end_date: String,
}

/// Struktur für die gesamte Ferien-JSON-Datei
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HolidayImportData {
    pub school_year: String,
    pub state: String,
    pub holidays: Vec<HolidayPeriod>,
}
